Logger/BaseLoggerMixins.cs

```diff
@@ -1,6 +1,39 @@
-﻿namespace Logger;
+using System;
+namespace Logger;
 
 public static class BaseLoggerMixins
 {
-    
+    public static void Error(this BaseLogger baseLogger, string message, params object[] args)
+    {
+        if (baseLogger is null) { throw new ArgumentNullException(); }
+
+        LogLevel logLevel = LogLevel.Error;
+        baseLogger.Log(logLevel, string.Format(message, args));
+    }
+
+    public static void Warning(this BaseLogger baseLogger, string message,
+                               params object[] args)
+    {
+        if (baseLogger is null) { throw new ArgumentNullException(); }
+
+        LogLevel logLevel = LogLevel.Warning;
+        baseLogger.Log(logLevel, string.Format(message, args));
+    }
+
+    public static void Information(this BaseLogger baseLogger, string message,
+                                   params object[] args)
+    {
+        if (baseLogger is null) { throw new ArgumentNullException(); }
```

Comment:
```suggestion
        if (baseLogger is null) { throw new ArgumentNullException(nameof(baselogger)); }
```
Make sure in all of your throws you are telling the exception what argument is unexpectedly null